.github/workflows/npm-cd.yml

```diff
@@ -151,6 +152,158 @@ jobs:
                   name: ${{ matrix.build.TARGET }}
                   path: ./node/bin
                   if-no-files-found: error
+            
+            - name: Reset repository
+              if: ${{ matrix.build.ARCH == 'arm64' }}
+              shell: bash
+              run: |
+                git reset --hard
+                git clean -xdf
+
+    checkout-self-hosted-runner:
+        if: github.repository_owner == 'aws'
+        needs: start-self-hosted-runner
+        runs-on: [self-hosted, Linux, ARM64]
+        steps:
+          - name: Setup self-hosted runner access
+            run: sudo chown -R $USER:$USER /home/ubuntu/actions-runner/_work/glide-for-redis
+          - name: Checkout
+            uses: actions/checkout@v4        
+    
+    publish-musl-binaries:
+        needs: [start-self-hosted-runner, checkout-self-hosted-runner]
+        if: github.repository_owner == 'aws'
+        name: Publish packages to NPM
+        runs-on: ${{ matrix.build.RUNNER }}
+        container:
+            image: node:alpine
+            options: --user root --privileged
+        strategy:
+            fail-fast: false
+            matrix:
+                build:
+                    - {
+                          OS: ubuntu,
+                          NAMED_OS: linux,
+                          ARCH: arm64,
+                          TARGET: aarch64-unknown-linux-musl,
+                          CONTAINER: "2_28",
+                          RUNNER: [self-hosted, Linux, ARM64],
+                      }
+                    - {
+                          OS: ubuntu,
+                          NAMED_OS: linux,
+                          ARCH: x64,
+                          TARGET: x86_64-unknown-linux-musl,
+                          RUNNER: ubuntu-latest,
+                      }
+        steps:
+            - name: Install dependencies
```

Comment:
It looks like Install dependencies, Checkout repository for x64, Checkout repository for ARM64 and Set up access are all parts of setting up the musl environment, and can be used by other actions as well (for example in build-and-test-linux-musl-latest). If you move it to a separate action (setup-linux-musl-container),  we can avoid separating the publish-musl-binaries job from the publish-binaries job, which will reduce the code duplication.

Reply:
You can add IMAGE and CONTAINER_OPTIONS fields to the original matrix in publish-binaries job:
                    - {
                          OS: ubuntu,
                          NAMED_OS: linux,
                          ARCH: x64,
                          TARGET: x86_64-unknown-linux-musl,
                          RUNNER: ubuntu-latest,
                          IMAGE: node:alpine,
                          CONTAINER_OPTIONS: '--user root --privileged',
                      }

It should be default to empty strings if these fields not provided in the matrix so adding 
```
        container:
            image: ${{ matrix.build.IMAGE }}
            options: ${{ matrix.build.CONTAINER_OPTIONS }}
```
to the general publish-binaries job should work, but please verify me